src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Carnivore) {
+      return;
+    }
+
+    if (target.hidden !== true && target instanceof Herbivore) {
+      target.health -= 50;
+
+      if (target.health <= 0) {
+        const animalIndex = Animal.alive.indexOf(target);
+
+        if (animalIndex !== -1) {
+          Animal.alive.splice(animalIndex, 1);
```

Comment:
Consider using a filter method to remove the dead animal from the 'Animal.alive' array to avoid nesting loops and improve code performance.